contracts/SunDAI.sol

```diff
@@ -0,0 +1,263 @@
+pragma solidity 0.5.2;
+
+import "../node_modules/openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "../node_modules/openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../node_modules/openzeppelin-solidity/contracts/access/roles/MinterRole.sol";
+
+
+contract SunDAI is IERC20, MinterRole {
+  using SafeMath for uint256;
+  
+  bytes32 public name;
+  bytes32 public symbol;
+  uint8 public decimals;
+
+  address public bridgeAddr;
+  IERC20 public dai;
+  mapping(address => uint256) public daiBalance;
+
+  constructor(address _daiAddr, address _bridgeAddr, bytes32 _name, bytes32 _symbol) public {
+    dai = IERC20(_daiAddr);
+    bridgeAddr = _bridgeAddr;
+    decimals = 18;
+    name = _name;
+    symbol = _symbol;
+  }
+
+  // 1. DAI is pulled into the contract, and record created
+  // 2. sunDAI are minted in amount of DAI deposited
+  // 3. sunDAI transfered to the Bridge
+  function _pullAndMint(address from, address to, uint256 value) internal {
+    dai.transferFrom(from, address(this), value);
```

Comment:
Shouldn't this be wrapped into a `require`. Otherwise I could just addresses that don't have DAI, the `dai.transferFrom` fails but I still get credited a balance.

Reply:
your are right. i'm assuming that the DAI token will throw, if there is not enough balance. My implementation actually does, but there are tokens that don't throw, and only return false. for these require is necessary.

Reply:
Actually never mind. https://etherscan.io/address/0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359#code DAI always returns true from what I can see here on L330. It throws on Over and Underflows.

Reply:
as long as it doesn't always return false 😬